### ISSUE TITLE
Ignore `S324` if an insecure hash function is called with `kwargs`

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#8165]
### Example code

```python
import hashlib
import sys

HASHLIB_KWARGS = (
    {"usedforsecurity": False} if sys.version_info >= (3, 9) else {}
)
hashlib.md5(**HASHLIB_KWARGS)
# ^^^^^^^^^
# S324
```

Can we ignore `S324` in this case?

**Comment by harupy** [comment]
Related to https://github.com/mlflow/mlflow/issues/10106

**Comment by zanieb** [comment]
Seems reasonable to me 👍

**Comment by Avasam** [comment]
This should definitely be well documented though. It wasn't immediately obvious to me why using kwargs makes this secure:

> Changed in version 3.9: All hashlib constructors take a keyword-only argument usedforsecurity with default value True. A false value allows the use of insecure and blocked hashing algorithms in restricted environments. False indicates that the hashing algorithm is not used in a security context, e.g. as a non-cryptographic one-way compression function.

Meaning if your dict doesn't include `usedforsecurity` then it stays default `True` and will be prevented in certain contexts. `usedforsecurity=False` assumes that you explicitely "approved" the use of insecure hash. **Unless the dict comes from user/external input**.

> Changed in version 3.9

Also meaning it should still raise for Python < 3.9

**Comment by charliermarsh** [comment]
I'm mixed on this, I'd only want to allow it if we can statically verify that the `**kwargs` resolve in this way, so as to avoid introducing false negatives for security-related warnings.

**Comment by harupy** [comment]
Thanks for the comments! I agree. `hashlib.md5(**kwargs)` doesn't necessarily mean kwargs has `usedforsecurity=False`. Closing.
